website/docs/components/form/key-value-inputs/partials/code/how-to-use.md

```diff
@@ -0,0 +1,351 @@
+      <F.Label>
+        Email address for invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.email}} />
+    </R.Field>
+    <R.Field as |F|>
+      <F.Label>
+        Name of invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.name}} />
+    </R.Field>
+    <R.DeleteRowButton />
+  </:row>
+</Hds::Form::KeyValueInputs>
+<hr />
+<Hds::Form::KeyValueInputs @isOptional={{true}} @data={{array
+    (hash name="Judith Maxene" email="j.maxene@randatmail.com")
+  }}>
+  <:header as |H|>
+    <H.Legend>Invite multiple users to your organization</H.Legend>
+  </:header>
+  <:row as |R|>
+    <R.Field as |F|>
+      <F.Label>
+        Email address for invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.email}} />
+    </R.Field>
+    <R.Field as |F|>
+      <F.Label>
+        Name of invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.name}} />
+    </R.Field>
+    <R.DeleteRowButton />
+  </:row>
+</Hds::Form::KeyValueInputs>
+```
+### Validation
+
+The Key Value Inputs can be validated at the fieldset or each individual field. You can choose which level is appropriate depending on the use case, but it is best practice to make any error message as specific as possible. Avoid putting the error on the fieldset if the issue is with a single field.
+
+To display an error on a field, set the `isInvalid` argument on the `[R].Field` yielded component, and use the `[F].Error` yielded component.
+
+```handlebars
+<Hds::Form::KeyValueInputs @data={{array
+    (hash name="Judith Maxene" email="")
+  }}>
+  <:header as |H|>
+    <H.Legend>Invite multiple users to your organization</H.Legend>
+    <H.HelperText>
+      Users without organization roles cannot view or edit anything inside their organization until project-level or workspace-level roles are assigned to them after they accept their invitation.
+    </H.HelperText>
+  </:header>
+  <:row as |R|>
+    <R.Field @isInvalid={{true}} as |F|>
+      <F.Label>
+        Email address for invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.email}} />
+      <F.Error>Email is required.</F.Error>
+    </R.Field>
+    <R.Field as |F|>
+      <F.Label>
+        Name of invitee
+      </F.Label>
+      <F.TextInput @value={{R.rowData.name}} />
+    </R.Field>
+    <R.DeleteRowButton />
+  </:row>
+  <:footer as |F|>
+    <F.AddRowButton @text="Add user" />
+  </:footer>
+</Hds::Form::KeyValueInputs>
+```
+
+To display an error for the whole group of inputs, use the `[F].Error` component yielded by the `:footer` named block.
+
+```handlebars
+<Hds::Form::KeyValueInputs @data={{array
+    (hash name="Judith Maxene" email="j.maxene@randatmail.com")
+  }}>
+  <:header as |H|>
+    <H.Legend>Invite multiple users to your organization</H.Legend>
+    <H.HelperText>
+      Users without organization roles cannot view or edit anything inside their organization until project-level or workspace-level roles are assigned to them after they accept their invitation.
```

Comment:
```suggestion
      Users without an organization role cannot view or edit anything inside their organization until a project-level or workspace-level role is assigned to them after they accept their invitation.
```

Reply:
Edited the above on the assumption that a user only need one appropriate-level role rather than needing to have multiple roles assigned to them.